lib/ex.coffee

```diff
@@ -150,25 +148,23 @@ class Ex
     if filePath.indexOf(' ') isnt -1
       throw new CommandError('Only one file name allowed')
 
-    deferred = Promise.defer()
-
-    editor = atom.workspace.getActiveTextEditor()
-    saved = false
-    if filePath.length isnt 0
-      fullPath = getFullPath(filePath)
-    if editor.getPath()? and (not fullPath? or editor.getPath() == fullPath)
-      # Use editor.save when no path is given or the path to the file is given
-      trySave(-> editor.save()).then(deferred.resolve)
-      saved = true
-    else if not fullPath?
-      fullPath = atom.showSaveDialogSync()
-
-    if not saved and fullPath?
-      if not force and fs.existsSync(fullPath)
-        throw new CommandError("File exists (add ! to override)")
-      trySave(-> saveAs(fullPath)).then(deferred.resolve)
-
-    deferred.promise
+    new Promise (resolve) ->
+
+      editor = atom.workspace.getActiveTextEditor()
+      saved = false
+      if filePath.length isnt 0
+        fullPath = getFullPath(filePath)
+      if editor.getPath()? and (not fullPath? or editor.getPath() == fullPath)
+        # Use editor.save when no path is given or the path to the file is given
+        trySave(-> editor.save()).then(resolve)
+        saved = true
+      else if not fullPath?
+        fullPath = atom.showSaveDialogSync()
+
+      if not saved and fullPath?
+        if not force and fs.existsSync(fullPath)
+          throw new CommandError("File exists (add ! to override)")
```

Comment:
This is the offending line - throwing an error from a Promise doesn't work like this, you'd have to use `reject` and then handle that... I'm not sure what the best way to go about this is.